### ISSUE TITLE
table read failure with ascii.commented_header format

### ISSUE THREAD

**Issue by PBarmby** [astropy/astropy#3407]
I have a file that looks like this:
```
# a b c d e f g h i j k l m n o
NGC14 108.13 -45.83 1.41 1.05 4.05E+20 12.8 T88 IB(s)m 10.0 2.4e+09 0.053 256 1.2e-14  ""  
NGC45 55.89 -80.67 4.26 2.95 2.24E+20 8.1 T92  SA(s)dm 8.0 3.8e+09 0.012 39 2.4e-15  ""  
NGC55 332.67 -75.74 16.18 2.81 1.72E+20 1.48 MM  SB(s)m 9.0 3.0e+09 0.076 57 1.6e-15  ""  
```
and I find that
```
from astropy.table import Table
test = Table.read('test',format='ascii')
```
works fine, while using `format='ascii.commented_header'` results in an ipython crash. This problem happens in astropy 1.0rc1 but *not* in 0.4.4. I'm using anaconda python under OS X 10.6.8.

**Comment by embray** [comment]
I got the same thing on Linux--this resulted in a segfault.  I wonder if this might be a duplicate of #3280, but I'm not sure.

**Comment by embray** [comment]
Also, no problem if I use `fast_reader=False` so it's probably a bug in the fast reader somewhere (not surprising since a segfault is unlikely to come from anywhere else...)

**Comment by embray** [comment]
I'm seeing
```
#0  0x00007fffeb304e51 in push (self=0x11b1100, c=78 'N', col=15) at astropy/io/ascii/src/tokenizer.c:115
115	    *self->col_ptrs[col]++ = c;
#1  0x00007fffeb305b61 in tokenize (self=0x11b1100, end=-1, header=0, num_cols=15)
    at astropy/io/ascii/src/tokenizer.c:535
```
I've never really looked at this code, but it's indexing past the length of col_ptrs.

**Comment by mdboom** [comment]
@embray: I was going to look at this, but if you're already doing so...  no problem.

**Comment by taldcroft** [comment]
Thoughts @mdmueller ?

**Comment by taldcroft** [comment]
Here is a smaller example that shows the problem:
```
In [4]: ascii.read(['a b', '1 ""', '2 ""'], format='fast_basic', guess=False)
/Users/aldcroft/anaconda/envs/astropy/bin/python.app: line 3:  6955 Segmentation fault: 11  /Users/aldcroft/anaconda/envs/astropy/python.app/Contents/MacOS/python "$@"
```

**Comment by mdmueller** [comment]
Interesting--this probably has something to do with the quotation marks in both examples. I'll look into this.

**Comment by embray** [comment]
Fixed by #3412 .

**Comment by embray** [comment]
Looking at this again, it seems unlikely now that this is directly related to #3280.
